src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -83,69 +104,76 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
      * values are deep copied.
      * @param[in] aOther The location object to copy.
      *
-     * @note If the locationName is empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
+     * @note This will do an EXACT copy of the other object. 
      */
     LocationStructureWrapper & operator=(const LocationStructureWrapper & aOther)
```

Comment:
Consider having an `operator==` as well that calls `IsEqual`

Reply:
Sadly, because it took so long to get comments on the ServiceArea Cluster specifications, the implementation/code branch is running significantly behind the latest updates to the spec.

With the current xml definitions, IsEqual checks the contents of the underlying LocationStruct, which is not the same as true equality which should be what the operator== indicates. And in the cluster implementation, IsEqual is what is needed, not operator==.  So for the moment it's better not to have it, as any use of it would almost certainly be wrong.

Again, the problem is that in terms of content
locationInfo.locationInfo { null } has the same content as locationInfo.locationInfo.Value()   { "", null, null }

Once all the xml and code is updated to reflect the very latest approved specification changes, all of the structure elements in the cluster will become non-nullable.

So there won't be any need in IsEquals to check for different topologies that are equal in content but not in form.
The special setter arguments will not be needed to select how to deal with the nullable structure.
An operator== based on IsEquals with the appropriate compare enum will be correct.
IsEquals is still necessary as there are a number of use cases where the code needs to compare all of the contents EXCEPT for the locationId, but that's clear based on the compare enum.

Reply:
OK, the point I am making is we need to be able to compare locations before/after, so that matter dirty attribute reporting can be done correctly at the right times on list changes.

Reply:
For now IsEquals with aConfig set to 0 is the appropriate function to use.

After the updated spec changes are pulled in and there are no more nullable structure elements, the operator== based on IsEquals can be added.